src/editor/Editor.js

```diff
@@ -425,16 +446,31 @@ define(function (require, exports, module) {
             tabSize                     : currentOptions[TAB_SIZE],
             readOnly                    : isReadOnly
         });
-
+        
         // Can't get CodeMirror's focused state without searching for
         // CodeMirror-focused. Instead, track focus via onFocus and onBlur
         // options and track state with this._focused
         this._focused = false;
 
         this._installEditorListeners();
 
+        // create the gutter space for the editor
+        var gutters = this._codeMirror.getOption("gutters").slice(0);
+        if (gutters.indexOf("interactive-gutter") === -1) {
+            gutters.unshift("interactive-gutter");
+            this._codeMirror.setOption("gutters", gutters);
+        }
+        this.on("gutterClicked",function(event, cm, lineIndex, gutterId){
+            self._gutterClick(cm, lineIndex, gutterId);
```

Comment:
The problem here is that you don't do any state checks to see if clicking should open or close an existing widget.  You have to track state so you don't re-open the widget every time, which is what you're doing now.

I think a better way to handle this click event would be to do exactly the same thing that `CMD-E` and `CTRL-E` do, which is to toggle the inline widget using the built-in command: https://github.com/adobe/brackets/blob/e78688aaa49af789726114721568637550bfaf86/src/command/Commands.js#L129  You can fire commands manually yourself.